assembly/src/parsers/ast/nodes.rs

```diff
@@ -241,3 +242,461 @@ pub enum Instruction {
     CallLocal(u32),
     CallImported(String),
 }
+
+pub trait Serializable: Sized {
+    fn to_bytes(&self) -> Vec<u8>;
+}
+
+impl Serializable for Vec<Node> {
+    fn to_bytes(&self) -> Vec<u8> {
+        let mut vec_bytes: Vec<u8> = Vec::new();
+
+        for node in self {
+            vec_bytes.append(&mut node.to_bytes());
+        }
+        let mut res: Vec<u8> = Vec::new();
+        res.append(&mut (vec_bytes.len() as u64).to_le_bytes().to_vec());
+        res.append(&mut vec_bytes);
+        res
+    }
+}
+
+impl Serializable for Node {
+    fn to_bytes(&self) -> Vec<u8> {
+        match self {
+            Self::Instruction(i) => {
+                let mut instruction_bytes: Vec<u8> = Vec::new();
+                instruction_bytes.push(0); // Instruction id = 0
+                instruction_bytes.append(&mut i.to_bytes());
+                instruction_bytes
+            }
+            Self::IfElse(left, right) => {
```

Comment:
`left` and `right` naming here is a bit confusing. I would use something like `if_clause` and `else_clause`.